jf_agent/main.py

```diff
@@ -592,21 +594,28 @@ def get_basic_jira_connection(config, creds):
 @diagnostics.capture_timing()
 @agent_logging.log_entry_exit(logger)
 def load_and_dump_github(config, endpoint_git_instance_info, git_conn):
-    write_file(
-        config.outdir,
-        'bb_users',
-        config.compress_output_files,
-        get_gh_users(git_conn, config.git_include_projects),
-    )
+    try:
+        write_file(
+            config.outdir,
+            'bb_users',
+            config.compress_output_files,
+            get_gh_users(git_conn, config.git_include_projects),
+        )
 
-    write_file(
-        config.outdir,
-        'bb_projects',
-        config.compress_output_files,
-        get_gh_projects(git_conn, config.git_include_projects, config.git_redact_names_and_urls),
-    )
+        write_file(
+            config.outdir,
+            'bb_projects',
+            config.compress_output_files,
+            get_gh_projects(git_conn, config.git_include_projects, config.git_redact_names_and_urls),
+        )
 
-    api_repos = None
+        api_repos = None
+    except Exception as e:
+        write_file_txt(
+            config.outdir,
+            'git_load_and_dump_exception',
+            e,
```

Comment:
1) This looks like it'll try to write out an `Exception` object as though it were text.  Is something implicitly string-ifying the exception in there somewhere?   
2) Text seems like the wrong format, perhaps? I'm imagining we may, over time, want more things in there (like, a boolean with an explicit "succeed/fail" rather than an implicit absence-of-a-thing meaning success; maybe a stack trace, maybe a string message, maybe other context over time), which makes me think of a structured `result.json` or some such thing

Reply:
1. yes - should be doing `repr(e)` to preserve the error along with its message
2. good point!